tests/jax/test_jax_3d_eldry.py

```diff
@@ -0,0 +1,1053 @@
+"""
+
+Script for testing the accuracy of the elastic dry friction implementation 
+that uses JAX - Compares AFT results to those from vector Jenkins
+
+Tests include:
+    1. Low and high normal displacement checks against Jenkins with different
+    values of Fs
+    2. Verify second harmonic of normal induces second harmonic of
+    tangential force when slipping is present (tests 3 and 5)
+    3. Verify if separated that all is zero.
+
+"""
+
+# Standard imports
+import numpy as np
+import sys
+import unittest
+
+# Python Utilities
+sys.path.append('../..')
+
+# # vectorized (non JAX version)
+# from tmdsimpy.jax.nlforces.jenkins_element import JenkinsForce 
+
+# JAX version for elastic dry friction
+from tmdsimpy.jax.nlforces import ElasticDryFriction3D 
+
+# JAX version for elastic dry friction
+from tmdsimpy.jax.nlforces import ElasticDryFriction2D 
+
+import tmdsimpy.utils.harmonic as hutils
+from tmdsimpy.vibration_system import VibrationSystem
+
+
+sys.path.append('..')
+import verification_utils as vutils
+
+
+###############################################################################
+###     Testing Class                                                       ###
+###############################################################################
+
+def run_comparison(obj, Unl, w, h, Nt, force_tol, df_tol, eldry2d, eldry3d,
+                   delta_grad=1e-5):
+    
+        dof = Unl.shape[0]
+        
+        # Create a mask tangent x and normal
+        txn = np.zeros(dof, dtype=bool)
+        txn[0::3] = True
+        txn[2::3] = True
+
+        tyn = np.zeros(dof, dtype=bool)
+        tyn[1::3] = True
+        tyn[2::3] = True
+         
+       
+        FnlH_vec, dFnldUH_vec, dFnldw_vec \
+            = eldry2d.aft(Unl[txn,:], w, h, Nt=Nt)
+        
+        FnlH, dFnldUH, dFnldw = eldry3d.aft(Unl, w, h, Nt=Nt)
+        
+        
+        FH_error = np.max(np.abs(FnlH[txn]-FnlH_vec))
+        
+        
+        obj.assertLess(FH_error, force_tol, 
+                        'Incorrect elastic dry friction force in X direction.')
+                
+        ###############
+        # Tangent - Tangent Gradient in X direction
+        dFH_error = np.max(np.abs(dFnldUH[np.ix_(txn,txn)]-dFnldUH_vec))
+        
+        obj.assertLess(dFH_error, df_tol, 
+                        'Incorrect Tangential AFT gradient in X direction.')
+        
+        
+        
+        FnlH_vec, dFnldUH_vec, dFnldw_vec \
+            = eldry2d.aft(Unl[tyn,:], w, h, Nt=Nt)
+        
+        FnlH, dFnldUH, dFnldw = eldry3d.aft(Unl, w, h, Nt=Nt)
```

Comment:
Why does the 3D force need to be re-evaluated here? This may contribute to this test being relatively slow.